service/web/adminserver.go

```diff
@@ -121,6 +129,19 @@ func (s adminServer) ModifyTopic(ctx context.Context, req *metrov1.AdminTopic) (
 		return nil, merror.ToGRPCError(uerr)
 	}
 
+	// Fetch Updated Topic
+	updatedTopic, err := s.topicCore.Get(ctx, m.Name)
+	if err != nil {
+		return nil, merror.ToGRPCError(err)
+	}
```

Comment:
why do we need to get this again..?
we can move  s.subscriptionCore.RescaleSubTopics(ctx, updatedTopic) call inside 
if int(req.NumPartitions) != existingTopic.NumPartitions block only, right..?

Reply:
Still we would need to fetch the updated topic right to pass it into the RescaleSubTopic arguments.